src/grow_worker/worker.py

```diff
@@ -1,61 +1,64 @@
+import io
+
 import jsonpickle
 import os
 import logging
-from typing import Optional
+from typing import Optional, Callable, Any
 from uuid import uuid4
 from pathlib import Path
 from celery import Celery
+from celery.signals import after_setup_logger
 from dataclasses import dataclass
-import subprocess
-
-from grow_runner.types import WorkFlowType
 
-LOGGER = logging.getLogger("grow_worker")
+from grow_worker.types import WorkFlowType, GROWProblem, get_problem_type
+from rtctools_heat_network.workflows import run_end_scenario_sizing
 
 app = Celery(
     "omotes",
     broker="amqp://user:bitnami@rabbitmq",
     backend="rpc://user:bitnami@rabbitmq",
     broker_connection_retry_on_startup=True,
+    worker_prefetch_multiplier=1,
+    task_acks_late=True,
+    task_reject_on_worker_lost=True,
+    # task_serializer="pickle",
+    # result_serializer="pickle",
+    # accept_content=["application/json", "application/x-python-serialize"],
 )
 
+logger = logging.getLogger(__name__)
+logging_string = None
+
+
+@after_setup_logger.connect
+def setup_loggers(logger, *args, **kwargs):
+    formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+
+    # add filehandler
+    global logging_string
+    logging_string = io.StringIO()
+    stream_handler = logging.StreamHandler(logging_string)
+    stream_handler.setLevel(logging.DEBUG)
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
 
 @app.task(name="optimizer-task", bind=True)
 def optimize(self, job_id: uuid4, esdl_string: str):
-    LOGGER.info("optimizer worker started")
-    return jsonpickle.encode(
-        run_rtc_calculation(
-            job_id,
-            esdl_string,
-            WorkFlowType.GROW_OPTIMIZER,
-        )
-    )
+    def update_progress(fraction: float, message: str) -> None:
```

Comment:
I prefer an object/class than a closure. Closure are implicit about which variables they capture. Also, closure uses each variable only by reference. If they update at any point (e.g. in a while-loop) they will change without the closure knowing it.